File: 프로그래머스/2/42584. 주식가격/주식가격.py
# 프로그래머스 42584 주식가격 LV.2
def solution(prices):
    from collections import deque
    answer = []
    prices = deque(prices)
    while prices:
        seconds = 0
        price = prices.popleft()

        for p in prices:
            if price <= p:
                seconds += 1
            else:
                seconds += 1
                break
        answer.append(seconds)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    Queue<Integer> q = new LinkedList<>();
    
    public int[] solution(int[] prices) {
        List<Integer> answer = new ArrayList<>();
        for (int p : prices) q.add(p);
        
        for (int i = 0; i < prices.length; i++) {
            int now = q.poll();
            
            if (q.isEmpty()) {
                answer.add(0);
                continue;
            }
            
            int tmp = 0;
            for (int nxt : q) {
                if (now > nxt) {
                    tmp++;
                    break;
                }
                tmp++;
            }
            
            answer.add(tmp);
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}